libraries/SITL/SIM_Vicon.cpp

```diff
@@ -204,6 +204,56 @@ void Vicon::update_vicon_position_estimate(const Location &loc,
             NULL, 0);
         msg_buf[msg_buf_index].time_send_us = time_send_us;
     }
+
+    // TO DO add Confidence Level
+    // # Confidence level value from T265: 0-3, remapped to 0 - 100: 0% - Failed / 33.3% - Low / 66.6% - Medium / 100% - High
+    // current_confidence_level = data.tracker_confidence * 33.33333f
+
+    // determine time, position, and angular deltas
+    static uint64_t prev_time_us = now_us + time_offset_us;
```

Comment:
Great that you got this working!

One small thing, could you move these statics into SIM_Vicon.h instead?  perhaps add a prefix to the variable names to make it clear that they're only used for the vision-position-delta part of the code.

Reply:
Updated so that the statics are in `SIM_Vicon.h`